### ISSUE TITLE
GEM needed when using future parser and puppetserver

### ISSUE THREAD

**Issue by mzahorik** [lynxman/hiera-consul#21]
See the discussion here:

https://tickets.puppetlabs.com/si/jira.issueviews:issue-html/SERVER-571/SERVER-571.html

TL;DR is that if you use the future parser and OSS puppetserver (>2.0?) it includes a new loader that breaks loading hiera backends found in modules.

A workaround is to the add the agent libdir to the puppetserver jRuby path like so for my installation:

   ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby, /opt/puppetlabs/puppet/cache/lib]

and then only restart the puppetserver after an agent run that syncs the module lib down to the agent cache directory.  Short term I suggest a documentation update of this workaround.

According to the thread, the correct go-forward solution is to convert hiera backends like this to a gem install, like hiera-eyaml does.

**Comment by mgla** [comment]
Since I only ever used the future parser, I did not notice this as a change of behaviour. Consequently, my [pull request](https://github.com/lynxman/hiera-consul/pull/22) for documentation improvements partially reflects this already.

**Comment by lynxman** [comment]
Thanks for this, will start working on it straight away

**Comment by lynxman** [comment]
Will integrate this with the puppet 5 hiera support ticket as it comes from the same place :)
